Makefile

```diff
@@ -52,39 +98,42 @@ kolibri.apk: p4a_android_distro preseeded_kolibri_home
 build_docker: Dockerfile
 	docker build -t android_kolibri .
 
-preseeded_kolibri_home: export KOLIBRI_HOME := src/preseeded_kolibri_home
-preseeded_kolibri_home: export PYTHONPATH := tmpenv
-preseeded_kolibri_home:
-	rm -r tmpenv 2> /dev/null || true
-	rm -r src/preseeded_kolibri_home 2> /dev/null || true
-	pip uninstall kolibri 2> /dev/null || true
-	pip install --target tmpenv whl/*.whl
-	tmpenv/bin/kolibri plugin enable kolibri.plugins.app
-	tmpenv/bin/kolibri start --port=0
-	sleep 1
-	tmpenv/bin/kolibri stop
-	sleep 1
-	yes yes | tmpenv/bin/kolibri manage deprovision
-	rm -r src/preseeded_kolibri_home/logs 2> /dev/null || true
-	rm -r src/preseeded_kolibri_home/sessions 2> /dev/null || true
-	rm -r src/preseeded_kolibri_home/process_cache 2> /dev/null || true
-	touch src/preseeded_kolibri_home/was_preseeded
-
 # Run the docker image.
 # TODO Would be better to just specify the file here?
 run_docker: build_docker
 	./scripts/rundocker.sh
 
-softbuild: project_info.json
-	pew build $(pew_release_flag) android ${ARCH}
-
 install:
 	adb uninstall org.learningequality.Kolibri || true 2> /dev/null
-	adb install dist/android/*$(ARM_VER)-debug-*.apk
+	adb install dist/*$(ARM_VER)-debug-*.apk
 
-run: install
-	adb shell am start -n org.learningequality.Kolibri/org.kivy.android.PythonActivity
-	sleep 1
+logcat:
 	adb logcat | grep -i -E "python|kolibr| `adb shell ps | grep ' org.learningequality.Kolibri$$' | tr -s [:space:] ' ' | cut -d' ' -f2` " | grep -E -v "WifiTrafficPoller|localhost:5000|NetworkManagementSocketTagger|No jobs to start"
 
-launch: softbuild run
+$(SDK)/cmdline-tools:
+	@echo "Downloading Android SDK build tools"
+	wget https://dl.google.com/android/repository/commandlinetools-$(PLATFORM)-7583922_latest.zip
+	unzip commandlinetools-$(PLATFORM)-7583922_latest.zip -d $(SDK)
+	rm commandlinetools-$(PLATFORM)-7583922_latest.zip
+
+sdk:
+	yes y | $(SDK)/cmdline-tools/bin/sdkmanager "platform-tools" --sdk_root=$(SDK)
+	yes y | $(SDK)/cmdline-tools/bin/sdkmanager "platforms;android-$(ANDROID_API)" --sdk_root=$(SDK)
+	yes y | $(SDK)/cmdline-tools/bin/sdkmanager "system-images;android-$(ANDROID_API);default;x86_64" --sdk_root=$(SDK)
+	yes y | $(SDK)/cmdline-tools/bin/sdkmanager "build-tools;30.0.3" --sdk_root=$(SDK)
+	yes y | $(SDK)/cmdline-tools/bin/sdkmanager "ndk;$(ANDROIDNDKVER)" --sdk_root=$(SDK)
+	@echo "Accepting all licenses"
+	yes | $(SDK)/cmdline-tools/bin/sdkmanager --licenses --sdk_root=$(SDK)
+
+setup:
+	$(MAKE) guard-ANDROID_HOME
+	mkdir -p $(SDK)
+	$(MAKE) $(SDK)/cmdline-tools
+	$(MAKE) sdk
+	@echo "Make sure to set the necessary environment variables"
+	@echo "export ANDROIDSDK=$(SDK)"
+	@echo "export ANDROIDNDK=$(SDK)/ndk/$(ANDROIDNDKVER)"
+	@echo "ANDROIDSDK=$(SDK)\nANDROIDNDK=$(SDK)/ndk/$(ANDROIDNDKVER)" > .env
```

Comment:
P4A has a `python-for-android` folder inside `~/.local`. I think personally, I'd rather clear caches a little bit more aggressively, to avoid caching issues (which are frustrating and hard to debug), rather than occasionally wait a little longer for a build.